site/docs/api/error-handling.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Error Handling Strategies
+description: Ways of coping with unexpected error in Hazelcast Jet.
+---
+
+There is unavoidable risk in running Jet jobs. They ingest data from
+external sources, the correctness and consistency of which can't be
+guaranteed. The processing they perform includes the running of user
+code, which is written by humans and, as such, is unavoidably imperfect.
+
+So even if we assume no bugs in the Jet framework itself, errors can and
+will happen. What options Jet offers for dealing with them is the topic
+of the next sections.
+
+## Scope
+
+A Jet cluster, at any given time, is running an arbitrary number of
+independent jobs in parallel. These jobs do not interact in any way, and
+the failure of one should not lead to any consequences for any other.
+When considering error handling, the basic unit of discussion will be
+the job, mainly what happens with it once it encounters a problem.
+
+The default way Jet reacts to an error is to fail the job that has
+produced it, but what one can do with such a failed job afterward is the
+interesting part.
+
+## Types of Errors
+
+Before we dig deeper, let's list some typical error one might encounter:
+
+* *corrupt data*: ex. user-provided code is parsing JSON lines, but some
+  of them are incorrectly formatted
+* *bugs in user code*: ex. user provided filtering lambda throws a
+  `NullPointerException` for certain input
+* *connection to external services lost*: Jet sink writing into an
+  `IMap` loses connection to the remote Hazelcast cluster hosting the
+  data
+
+## Specific Solutions
+
+Some concrete types of error have solutions specifically tailored to
+address them.
+
+### In-Memory Data Structures
+
+Jet comes out of the box with some [in-memory distributed data
+structures](data-structures) which can be used as a data source or a
+sink. This are hosted by Hazelcast clusters, which can be the same as
+the one Jet runs on or separate, remote ones.
+
+In this case, all error handling (for example, communications failures
+with the remote cluster, but not just) is handled transparently to Jet,
```

Comment:
"but not just *those*"
"to Jet" -> "by Jet"

Reply:
I don't agree with the "to -> by" change. I wanted to say that Jet is not aware of the error, and for that "transparent to Jet" seems the right expression.

Reply:
Instead of this confusion just use the active voice and say explicitly who is the actor.

Reply:
@jbartok you're right, I missed it.